video/out/wayland_common.c

```diff
@@ -32,17 +32,468 @@
 #include "osdep/timer.h"
 #include "wayland_common.h"
 #include "win_state.h"
+#include <drm_fourcc.h>
 
 // Generated from wayland-protocols
 #include "generated/wayland/idle-inhibit-unstable-v1.h"
 #include "generated/wayland/presentation-time.h"
 #include "generated/wayland/xdg-decoration-unstable-v1.h"
 #include "generated/wayland/xdg-shell.h"
+#include "generated/wayland/linux-dmabuf-unstable-v1.h"
+#include "generated/wayland/viewporter.h"
 
 #if WAYLAND_VERSION_MAJOR > 1 || WAYLAND_VERSION_MINOR >= 20
 #define HAVE_WAYLAND_1_20
 #endif
 
+const char* vo_wayland_drm_format_string(uint drm_format) {
```

Comment:
Yeah I think I'm going to NACK this function. It's not your fault or anything; this is the only way to map the format to a name. But it shouldn't be mpv's job to do this. DRM should provide this API (I don't think there is one but haven't checked in a while). I'm OK with the tradeoff of having less informative logs.

Reply:
This could be moved to `drm_common.c` at least. Digging around, it seems like the kernel just use prink nowadays so I guess there's no way around something like this.